### PR TITLE
Fix wrong index rule binding for Measure

### PR DESCRIPTION
**Pull request by lujiajing1126**
Fix https://github.com/apache/skywalking/issues/8979

Previously the `IndexRuleBinding` refers to a wrong target kind which would cause index rule reload failure in BanyanDB. Test cases are added.

Besides, `ID` type is added for measure support.

### CODE REVIEW

File: src/main/java/org/apache/skywalking/banyandb/v1/client/TagAndValue.java [hunk 1]
```diff
@@ -104,6 +104,26 @@ public static TagAndValue<String> newStringTag(final String tagName, final Strin
         return new StringTagPair(tagName, value);
     }
 
+    @EqualsAndHashCode(callSuper = true)
+    public static class IDTagPair extends TagAndValue<String> {
```
Comment (wu-sheng) [inline src/main/java/org/apache/skywalking/banyandb/v1/client/TagAndValue.java:108]: Just a confirmation, ID column requires a tag name too? Could you share how ID tag works?(Just for helping me catching up context).

Comment (lujiajing1126) [reply src/main/java/org/apache/skywalking/banyandb/v1/client/TagAndValue.java:108]: Emmmm, I've checked the code again. You're right. According to the implementation, it has to be named as `id` which is required by the hardcoded name here https://github.com/apache/skywalking-banyandb/blob/49a23ae9bdcc65ae7028617a3bf86bb243ffec64/banyand/metadata/schema/measure.go#L172

But we actually don't have any kind of mandatory validation for this name elsewhere. On the client-side, I should have hidden the low-level impl here.

As I understood, 

1. `ID` is the default sharding key if no other sharding key is specified (e.g. `entity_id`),
2. `ID` is indexed automatically which allows us to search by ID directly,

@hanahmily any other comment?

Comment (wu-sheng) [reply src/main/java/org/apache/skywalking/banyandb/v1/client/TagAndValue.java:108]: > But we actually don't have any kind of mandatory validation for this name elsewhere. On the client-side, I should have hidden the low-level impl here.

I am thinking whether we should not expose `tag` key, but hard coded in client?

> ID is the default sharding key if no other sharding key is specified (e.g. entity_id),

Entity ID should have `sharding key` annotation by the kernel, if there is not, we should add it. Default sharding key is a dangerous mechanism, let's make it mandatory.

Comment (lujiajing1126) [reply src/main/java/org/apache/skywalking/banyandb/v1/client/TagAndValue.java:108]: > I am thinking whether we should not expose `tag` key, but hard coded in client?

Now it is almost hidden. It now works like a primary key. Users only need to `setID` in the write request or directly give an `idValue` via `MeasureQuery.andWithID`.

Now the only place, OAP may still touch the `ID` literal is for sharding keys.

But as you've suggested above, we have to explicitly specify the sharding keys.

> Entity ID should have sharding key annotation by the kernel, if there is not, we should add it. Default sharding key is a dangerous mechanism, let's make it mandatory.

I suppose most of the measure models in OAP have `entity_id` as the sharding key. But what about `*Traffic` models?

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**